curriculum/challenges/english/25-front-end-development/review-web-performance/6724e18296fa40173cc2437c.md

```diff
@@ -9,7 +9,137 @@ dashedName: review-web-performance
+- **Leverage Browser Caching**: Caching allows browsers to store parts of your website locally, reducing load times for returning visitors.
+- **Minify and Compress Files**: Reducing the size of your files can lead to quicker downloads. This includes reducing the size of transmitted files and minifying CSS and JavaScript files.
+
+## Improving "time to usable"
+
+- **Definition**: "time to usable" is the interval from when a user requests a page to when they can meaningfully interact with it. To improve "time to usable" you can lazy load your asset or minimize render-blocking resources.
+
+## Key Metrics for Measuring Performance
+
+- **First Contentful Paint or FCP**: First Contentful Paint or FCP measures how quickly the first piece of content—text or image—appears on the screen. A good FCP is regarded as a time below 1.8 seconds, and a poor FCP is above 3 seconds. You can check your FCP using Chrome DevTools and checking the performance tab
+- **Total Blocking Time**: This shows how long the main thread is blocked by heavy JavaScript tasks. If Total Blocking Time(TBT) is high, users experience sluggish interactions. To improve TBT, break up long tasks and defer non-essential scripts.
+- **Bounce Rate**: This is the percentage of visitors who leave without interacting. if your site has High bounce rates - it might be because your page is too slow.
+- **Unique Users**: This metric tracks how many individual visitors come to your site. To view the Bounce Rate and Unique Users, you can use Google Analytics. It will allow you to monitor these metrics and improve engagement.
+
+## Common Performance Measurement Tools
+
+- **Chrome DevTools**: Chrome DevTools is a built-in tool inside Google Chrome that lets you analyze and debug performance in real-time. DevTools will show loading times, CPU usage, and render delays. It's especially useful for measuring First Contentful Paint, or FCP—which we know by now- is how fast a user sees the first visible content. If your website feels slow, DevTools will help you spot the bottlenecks. 
+- **Lighthouse**: This is an automated tool that checks performance, SEO, and accessibility. 
+- **WebPageTest**: This tools lets you test how your site loads from different locations and devices. This tool gives you a detailed breakdown of your site's Speed Index, Total Blocking Time, and other key performance metrics. If you want to know how real users experience your site globally, WebPageTest is the tool for that.
+- **PageSpeed Insights**: This tool analyzes your website and suggests quick improvements for both mobile and desktop. It will tell you what's slowing your site down and give specific recommendations—like optimizing images, removing render-blocking scripts, and reducing server response times. PageSpeed Insights is a fast and easy way to check how Google sees your site's performance.
```

Comment:
```suggestion
- **PageSpeed Insights**: This tool analyzes your website and suggests quick improvements for both mobile and desktop. It will tell you what's slowing your site down and give specific recommendations like optimizing images, removing render-blocking scripts, and reducing server response times. PageSpeed Insights is a fast and easy way to check how Google sees your site's performance.
```